featurebyte/session/base.py

```diff
@@ -490,7 +555,12 @@ async def get_async_query_stream(
         try:
             # execute in separate thread
             await to_thread(
-                cursor.execute, timeout, query, **self.get_additional_execute_query_kwargs()
+                self._execute_query,
+                timeout,
+                lambda *args, **kwargs: self._try_cancel_query(cursor, query),
```

Comment:
Are `args` and `kwargs` needed here? If not, might be better to make the lambda have no argument and call the handler without passing anything in.

Reply:
probably not. Will remove for now - we can update this later if we need the callback to use any other information not available before code is run (e.g. exception)